mirrord/cli/src/container/sidecar.rs

```diff
@@ -0,0 +1,158 @@
+            .await?
+            .ok_or_else(|| {
+                ContainerError::UnsuccesfulCommandOutput(
+                    format_command(&sidecar_container_spawn),
+                    "stdout and stderr were empty".to_owned(),
+                )
+            })?;
+
+        Ok(Sidecar {
+            container_id,
+            runtime_binary,
+        })
+    }
+
+    pub fn as_network(&self) -> String {
+        let Sidecar { container_id, .. } = self;
+        format!("container:{container_id}")
+    }
+
+    #[tracing::instrument(level = Level::TRACE)]
+    pub async fn start(&self) -> Result<(SocketAddr, SidecarLogs), ContainerError> {
+        let mut command = Command::new(&self.runtime_binary);
+        command.args(["start", "--attach", &self.container_id]);
+
+        let mut child = command
+            .stdin(Stdio::null())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()
+            .map_err(ContainerError::UnableToExecuteCommand)?;
+
+        tracing::warn!(?child, "spawned watch for child");
+
+        let mut stdout =
+            BufReader::new(child.stdout.take().expect("stdout should be piped")).lines();
+        let stderr = BufReader::new(child.stderr.take().expect("stdout should be piped")).lines();
+
+        let first_line = tokio::time::timeout(Duration::from_secs(30), async {
+            stdout.next_line().await.map_err(|error| {
+                ContainerError::UnableReadCommandStdout(format_command(&command), error)
+            })
+        })
+        .await
+        .map_err(|_| {
+            ContainerError::UnsuccesfulCommandOutput(
+                format_command(&command),
+                "timeout reached for reading first line".into(),
+            )
+        })??
+        .ok_or_else(|| {
+            ContainerError::UnsuccesfulCommandOutput(
+                format_command(&command),
+                "unexpected EOF".into(),
+            )
+        })?;
+
+        let internal_proxy_addr: SocketAddr = first_line
+            .parse()
+            .map_err(ContainerError::UnableParseProxySocketAddr)?;
+
+        Ok((internal_proxy_addr, SidecarLogs { stdout, stderr }))
+    }
+}
+
+pin_project! {
+    #[derive(Debug)]
+    pub(crate) struct SidecarLogs {
+        #[pin]
+        pub stdout: Lines<BufReader<ChildStdout>>,
+        #[pin]
+        pub stderr: Lines<BufReader<ChildStderr>>,
+    }
+}
+
+impl Stream for SidecarLogs {
+    type Item = Result<String, std::io::Error>;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this = self.project();
+
+        if let result @ Poll::Ready(Some(_)) = this
+            .stdout
+            .poll_next_line(cx)
+            .map(|result| result.transpose())
+        {
+            return result;
+        }
+
+        this.stderr
+            .poll_next_line(cx)
+            .map(|result| result.transpose())
```

Comment:
Maybe this is a nitpick - which logs does the sidecar send to stdout and which ones to stderr? Is it ok to be biased towards stdout?
If you think it's ok as-is that's fine with me. If you think we should change it, maybe [tokio's `merge`](https://docs.rs/tokio-stream/0.1.17/tokio_stream/trait.StreamExt.html#method.merge) is helpful.

Reply:
Ok this entire thing is not needed because of `LinesStream`